src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -37,11 +37,55 @@ namespace CDASimAdapter{
         PLOG(logINFO) << "CARMA-Simulation connection is successful!" << std::endl;
         return _connected;
     }
-    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint simulation_registration_port, 
+    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint infrastructure_id, const uint simulation_registration_port, 
                                 const std::string &local_ip,  const uint time_sync_port, const uint v2x_port, 
-                                const WGS84Point &location) {
-        // TODO: Standup UDP Server and Client for registration and implement handshake
-        return false;                                
+                                const WGS84Point &location) 
+    {
+        
+        //Get remote endpoint
+        try
+        {
+            remote_udp_ep_ = boost::asio::ip::udp::endpoint(boost::asio::ip::address::from_string(simulation_ip), simulation_registration_port);
+        }
+        catch(std::exception e)
+        {
+            PLOG(logERROR) << "Encountered runtime error when resolving remote udp end point given: " << e.what() << std::endl;
+            throw e;
```

Comment:
I don't think we need to catch and exception, log it and then rethrow it. I think the better thing would probably be to return false here. Please if its not to much trouble, we should be being consistent by using the UDPClient class instead of  using a different UDP client implementation

Reply:
Will do